doc/extensions.rst

```diff
@@ -57,6 +62,75 @@ Once the new extension class has been defined, the openml-python module to
+* Serialization and De-serialization (required)
+  * :meth:`flow_to_model`: deserializes the OpenML Flow into a model (if the
+    library can indeed handle the flow). This method has an important interplay
+    with :meth:`model_to_flow`. 
+    Running these two methods in succession should result in exactly the same
+    model (or flow). This property can be used for unit testing (e.g., build a
+    model with hyperparameters, make predictions on a task, serialize it to a flow,
+    deserialize it back, make it predict on the same task, and check whether the
+    predictions are exactly the same.)
+    The example in the scikit-learn interface might seem daunting, but note that
+    here some complicated design choices were made, that allow for all sorts of
+    interesting research questions. It is probably good practice to start easy. 
+  * :meth:`model_to_flow`: The inverse of :meth:`flow_to_model`. Serializes a 
+    model into an OpenML Flow. The flow should preserve the class, the library
+    version, and the tunable hyperparameters. 
+  * :meth:`get_version_information`: Return a tuple with the version information
+    of the important libraries. 
+  * :meth:`create_setup_string`: No longer used, and will be deprecated soon. 
+* Performing runs (required)
+  * :meth:`is_estimator`: Gets as input a class, and checks whether it has the 
+    status of estimator in the library (typically, whether it has a train method
+    and a predict method).
+  * :meth:`seed_model`: Sets a random seed to the model. 
+  * :meth:`_run_model_on_fold`: One of the main requirements for a library to
+    generate run objects for the OpenML server. Obtains a train split (with
+    labels) and a test split (without labels) and the goal is to train a model
+    on the train split and return the predictions on the test split.
+    On top of the actual predictions, also the class probabilities should be
+    determined. 
+    For classifiers that do not return class probabilities, this can just be the
+    hot-encoded predicted label.
+    The predictions will be evaluated on the OpenML server.
+    Also, additional information can be returned, for example, user-defined
+    measures (such as runtime information, as this can not be inferred on the
+    server). 
+    Additionally, information about a hyperparameter optimization trace can be
+    provided. 
+  * :meth:`obtain_parameter_values`: Obtains the hyperparameters of a given
+    model and the current values. Please note that in the case of a hyperparameter
+    optimization procedure (e.g., random search), you only should return the
+    hyperparameters of this procedure (e.g., the hyperparameter grid, budget,
+    etc) and that the chosen model will be inferred from the optimization trace. 
+  * :meth:`check_if_model_fitted`: Check whether the train method of the model
+    has been called (and as such, whether the predict method can be used).
+* Hyperparameter optimization (optional)
```

Comment:
A small addition that can be added here is
The optional methods have to be initialized as an empty function example:
def instantiate_model_from_hpo_class():
    "This is just an empty function to avoid errors"

It can be confusing for some users to get errors that this optional method is not implemented. So here we can add that though these methods are optional they should still be present in the extension as an empty function.

Reply:
What do you mean exactly? I would just suggest users to raise a `NotImplementedError` (since that method will only be invoked externally)

Reply:
In the end, you still need these optional methods to exist in the extension, While using the extension openml-python does require the optional methods to exist too(at least that was the issue I faced while developing openml-tensorflow).  So I just defined the functions without definitions like here: https://github.com/openml/openml-tensorflow/blob/master/openml_tensorflow/extension.py

But it was confusing for me at first when I was getting those errors, so I think it will be helpful to add a message regarding that here(that we still need to define these methods as empty methods if we don't want to implement them).

Reply:
> In the end, you still need these optional methods to exist in the extension

Are you sure? Since the extensin inherits from the `Extension` interface, we technically do not need to implement it in the specific right? 

Just to be sure, I added: 
> Note that although this method is optional, for the extension to work it
> needs to exist. If this function is not required, we recommend to have an
> empty function that raises a `NotImplementedError` with a useful error
> message.